include/IECoreScene/SceneAlgo.h

```diff
@@ -0,0 +1,64 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#include "IECoreScene/Export.h"
+
+#include "IECoreScene/SceneInterface.h"
+
+#include <map>
+#include <string>
+
+namespace IECoreScene
+{
+	namespace SceneAlgo
+	{
+
+		enum CopyFlags
+		{
+			None = 0,
+			Bounds= 1 << 0,
+			Transforms= 1 << 1,
+			Attributes = 1 << 2,
+			Tags = 1 << 3,
+			Sets = 1 << 4,
+			Objects = 1 << 5,
+			All = Bounds | Transforms | Attributes | Tags | Sets | Objects
+		};
+
+		typedef std::map<std::string, size_t> CopyStats;
+
+		/// copy from one scene to another.
+		IECORESCENE_API CopyStats copy( ConstSceneInterfacePtr src, SceneInterfacePtr dst, int startFrame, int endFrame, float frameRate, int threadCount, unsigned int flags );
```

Comment:
I think this review is turning into an exact copy of the IndexedIO one! I feel like this should be split out into several methods with specific purposes, along the same lines as I suggested for IndexedIOAlgo.

Might also be useful to have a version that was SampledSceneInterface-savvy, so we didn't need to provide the time information, but that could wait until we have a direct need.